### PR TITLE
Include Huggingface model to E2E models tests

### PR DESCRIPTION
**Pull request by sgrebnov**
# 🗣 Description

Include Huggingface model to E2E models tests

Example run: https://github.com/spiceai/spiceai/actions/runs/12206817992

## 🔨 Related Issues

Part of https://github.com/spiceai/spiceai/issues/3728

## 🤔 Concerns

We can continue using hosted macos runners for build (support build with `metal` option) and use self hosted runners for tests only. Switched to own runners as they are faster, we can switch back to hosted runners once we have more test workload that must be run on spice macos runners.

### CODE REVIEW

File: .github/workflows/e2e_test_ci_models.yml [hunk 1]
```diff
@@ -6,10 +6,10 @@ on:
       - trunk
       - release-*
     paths-ignore:
-      - 'docs/**'
-      - 'README.md'
-      - 'version.txt'
-      - 'acknowledgements.md'
+      - "docs/**"
```
Comment (lukekim) [inline .github/workflows/e2e_test_ci_models.yml]: Fix your prettier settings to single quote :-)

File: .github/workflows/e2e_test_ci_models.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ jobs:
             target_os: "linux"
             target_arch: "x86_64"
           - name: "macOS aarch64 (Apple Silicon)"
-            builder: "macos-14"
+            builder: [self-hosted, macOS]
```
Comment (lukekim) [inline .github/workflows/e2e_test_ci_models.yml]: Can we rename this from `self-hosted` to `spice-hosted`?

File: .github/workflows/e2e_test_ci_models.yml [hunk 3]
```diff
@@ -194,4 +202,73 @@ jobs:
           killall spice || true
           cat spice.log
 
-  
\ No newline at end of file
+  test_hf_model:
+    name: "huggingface model (${{ matrix.target.target_os }}-${{ matrix.target.target_arch }})"
```
Comment (Sevenannn) [inline .github/workflows/e2e_test_ci_models.yml]: Quotes are not unified, use `"` or `'` instead of both

File: test/models/spicepod_hf.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          trim_whitespace: false  
+        use: intfloat_e5
+    
+models:
+  - name: hf_local_model
+    from: huggingface:huggingface.co/microsoft/Phi-3-mini-4k-instruct
+    params:
+      model_type: phi3
+      spice_tools: auto
+      system_prompt: |
```
Comment (Sevenannn) [inline test/models/spicepod_hf.yml:61]: Just curious about how the system prompts are generated, they are well written! Did you write it yourself, or there's other tools / reference used.

File: .github/workflows/e2e_test_ci_models.yml (hunk 1 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Configured - thank you

File: .github/workflows/e2e_test_ci_models.yml (hunk 2 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Updated to use existing label `spiceai-macos` (created yesterday after PR) - thank you

File: .github/workflows/e2e_test_ci_models.yml (hunk 3 above)
Comment (sgrebnov) [reply .github/workflows/e2e_test_ci_models.yml]: Configured - than you

File: test/models/spicepod_hf.yml (hunk 4 above)
Comment (sgrebnov) [reply test/models/spicepod_hf.yml:61]: Was taken from Spice with the OpenAI SDK sample:
https://github.com/spiceai/samples/blob/b782524f5850bc4495d8b52d618081dcd1080c19/spice_openai_sdk/spicepod.yaml

File: test/models/spicepod_hf.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  - from: github:github.com/spiceai/spiceai/issues
+    name: issues
+    description: GitHub Issues for the Spice.ai project (github.com/spiceai/spiceai)
+    metadata:
+      instructions: Always provide reference links. Use the primary key as `issue_id` to generate reference links.
+      reference_url_template: https://github.com/spiceai/spiceai/issues/<issue_id>
+    params:
+      github_token: ${secrets:GITHUB_TOKEN}
+      github_query_mode: "search"
```
Comment (Sevenannn) [inline test/models/spicepod_hf.yml]: This quotes in this file needs to be updated as well

Comment (sgrebnov) [reply test/models/spicepod_hf.yml]: Updated, thanks

**Review by Sevenannn — APPROVED** [review]

**Pull request merged**